test/tests/functional/pbs_mom_dynamic_resource_ext.py

```diff
@@ -0,0 +1,212 @@
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestMomDynRes(TestFunctional):
+    def create_mom_dyn_resc(self, resc_name,
+                            resc_type, resc_flag,
+                            script_body):
+        """
+        helper function for create mom_dyn_res.
+        """
+        attr = {"type": resc_type, "flag": resc_flag}
+        self.server.manager(MGR_CMD_CREATE, RSC, attr,
+                            id=resc_name, expect=True)
+
+        (fd, fp) = self.du.mkstemp(prefix="mom_dyn_resc", suffix=".scr",
+                                   body=script_body)
+        self.du.chmod(path=fp, mode=0755)
+        os.close(fd)
+
+        mom_config_str = "!" + fp
+        self.mom.add_config({resc_name: mom_config_str})
+        self.scheduler.set_sched_config({'mom_resources': '"foo"'},
+                                        validate=True)
+        self.scheduler.add_resource(resc_name)
+
+    def test_t1(self):
+        """
+        Test for host level consumable size resource
+        with string mom dynamic value
+        """
+        resc_name = "foo"
+        self.create_mom_dyn_resc(resc_name, "size", "nh", "/bin/echo abc")
+
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Submit a job that requests mom dynamic resource
+        attr = {"Resource_List." + resc_name: 1}
+        j = Job(TEST_USER, attrs=attr)
+        jid = self.server.submit(j)
+
+        # The job shouldn't run
+        self.server.expect(JOB, {'job_state': 'Q'}, id=jid)
+
+        # Check for the expected log message in scheduler logs
+        c = "Insufficient amount of resource: foo (1kb != (null))"
```

Comment:
Any time you see '(null)' in a string from PBS, then it is a bug.  It means a NULL pointer was passed into printf().  This is not something you should be testing for in a PTL test.